%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% This is just an example/guide for you to refer to when submitting manuscripts to Frontiers, it is not mandatory to use Frontiers .cls files nor frontiers.tex  %
% This will only generate the Manuscript, the final article will be typeset by Frontiers after acceptance.
%                                              %
%                                                                                                                                                         %
% When submitting your files, remember to upload this *tex file, the pdf generated with it, the *bib file (if bibliography is not within the *tex) and all the figures.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% Version 3.4 Generated 2022/06/14 %%%
%%% You will need to have the following packages installed: datetime, fmtcount, etoolbox, fcprefix, which are normally inlcuded in WinEdt. %%%
%%% In http://www.ctan.org/ you can find the packages and how to install them, if necessary. %%%
%%%  NB logo1.jpg is required in the path in order to correctly compile front page header %%%

\documentclass[utf8]{FrontiersinHarvard} % for articles in journals using the Harvard Referencing Style (Author-Date), for Frontiers Reference Styles by Journal: https://zendesk.frontiersin.org/hc/en-us/articles/360017860337-Frontiers-Reference-Styles-by-Journal

\usepackage{url,hyperref,lineno,microtype,subcaption}
\usepackage[onehalfspacing]{setspace}

\linenumbers

% Leave a blank line between paragraphs instead of using \\

\def\keyFont{\fontsize{8}{11}\helveticabold }
\def\firstAuthorLast{Lee {et~al.}} %use et al only if is more than 1 author
\def\Authors{Jover Lee\,$^{1}$, John Huddleston\,$^{1}$, Allison Black\,$^{2}$, Thomas R. Sibley\,$^{1}$, Richard A. Neher\,$^{3,4}$, Trevor Bedford\,$^{1,5}$, and James Hadfield\,$^{1,*}$}
% Affiliations should be keyed to the author's name with superscript numbers and be listed as follows: Laboratory, Institute, Department, Organization, City, State abbreviation (USA, Canada, Australia), and Country (without detailed address information such as city zip codes or street names).
% If one of the authors has a change of address, list the new address below the correspondence details using a superscript symbol and use the same symbol to indicate the author in the author list.
\def\Address{$^{1}$Vaccine and Infectious Disease Division, Fred Hutchinson Cancer Center, Seattle, WA, USA \\
  $^{2}$Chan Zuckerberg Initiative, CA, San Francisco, CA, USA \\
  $^{3}$Biozentrum, Universität Basel, Switzerland \\
  $^{4}$Swiss Institute of Bioinformatics, Switzerland \\
  $^{5}$Howard Hughes Medical Institute, Seattle, WA, USA}
% The Corresponding Author should be marked with an asterisk
% Provide the exact contact address (this time including street name and city zip code) and email of the corresponding author
\def\corrAuthor{James Hadfield}
\def\corrEmail{}

\begin{document}
\onecolumn
\firstpage{1}

\title[Joint visualization of flu data]{Joint visualization of seasonal influenza serology and phylogeny to inform vaccine strain selection}

\author[\firstAuthorLast ]{\Authors} %This field will be automatically populated
\address{} %This field will be automatically populated
\correspondance{} %This field will be automatically populated

\extraAuth{}% If there are more than 1 corresponding author, comment this line and uncomment the next one.
%\extraAuth{corresponding Author2 \\ Laboratory X2, Institute X2, Department X2, Organization X2, Street X2, City X2 , State XX2 (only USA, Canada and Australia), Zip Code2, X2 Country X2, email2@uni2.edu}

\maketitle

\begin{abstract}

\section{}

Seasonal influenza vaccines must be updated regularly to account for mutations that allow influenza viruses to escape our existing immunity.
A successful vaccine should represent the genetic diversity of recently circulating viruses and induce antibodies that effectively prevent infection by those recent viruses.
Thus, linking the genetic composition of circulating viruses and the serological experimental results measuring antibody efficacy is crucial to the vaccine design decision.
Historically, genetic and serological data have been presented separately in the form of static visualizations of phylogenetic trees and tabular serological results to identify vaccine candidates.
To simplify this decision-making process, we have created an interactive tool for visualizing serological data that has been integrated into Nextstrain’s real-time phylogenetic visualization framework, Auspice.
We show how the combined interactive visualizations may be used by decision-makers to explore the relationships between complex data sets for both prospective vaccine strain selection and retrospectively exploring the performance of vaccine strains.

\tiny
 \keyFont{ \section{Keywords:} keyword, keyword, keyword, keyword, keyword, keyword, keyword, keyword} %All article types: you may provide up to 8 keywords; at least 5 are mandatory.
\end{abstract}

\section{Introduction}

A primary component of seasonal influenza A/H3N2 evolution is the ability of viruses to acquire mutations that allow them to escape antibodies from previous infections.
This process, known as antigenic drift, changes the appearance of viral surface proteins hemagglutinin (HA) and neuraminidase (NA).
Viruses balance escape from antibodies with the maintenance of their protein functions.
The HA surface protein allows viruses to bind to the surface of new host cells and initiate infection.
When antibodies bind to HA, they can prevent viruses from binding and infecting cells.
Viruses that acquire mutations to HA that prevent antibodies from binding but do not disrupt the ability of HA to bind to host cells should be able to infect hosts.

Experimental measurements of antigenic drift allow researchers to quantify how well viruses with different HA mutations can escape detection by antibodies.
Until recently, the most reliable of these experimental measurements were hemagglutination inhibition (HI) assays \citep{hirst1943studies}.
In these assays, researchers place red blood cells into a multi-well plate and add one test virus per well.
When a well contains only blood cells and virus, the virus binds to the blood cells causing them to agglutinate into a wide, red dot that fills the well.
Next, researchers add two-fold dilutions of antisera from naive ferrets that were infected by a single reference virus.
When the antisera contains enough antibodies to effectively bind the virus, the blood cells do not agglutinate and instead sink to the bottom of the well in a small red dot.
The highest dilution of antisera required to inhibit agglutination provides the ``titer'' measurement between the reference and test viruses.
When the test virus is the same as the reference virus, the assay provides an autologous titer measurement.
When the test and reference viruses differ, the assay provides a heterologous titer measurement.

Researchers report titer measurements as both raw and normalized values.
Raw titer measurements represent the denominator associated with the minimum dilution required to inhibit agglutination.
These two-fold dilution series have raw measurements like 80, 160, 320, etc. such that lower numbers represent the presence of more antisera in the dilution.
These raw measurements can also be represented more conveniently on a $\log_{2}$ scale \citep{Smith:2004jc,Bedford:2014bf}.
Antisera vary in their potency such that some sera always require higher or lower dilutions to inhibit agglutination regardless of the test virus.
For example, a low potency antiserum requires a lower dilution to inhibit agglutination by the same reference virus resulting in a low autologous titer.
To account for this variable potency, we normalize titer measurements by subtracting the $\log_{2}$ titer between a test and reference virus (the heterologous titer) from the $\log_{2}$ titer between the reference virus and its own antisera (the autologous titer).
The resulting normalized titers enable comparisons of antigenic distances across reference viruses.
Importantly, viruses with a $\log_{2}$ distance greater than 2 are considered antigenically distinct for the purposes of deciding updates to vaccine composition \citep{Katz2011}.

To understand broad patterns of antigenic drift beyond simple pairwise relationships, antigenic distance must be summarized by statistical or visual representations.
For example, the method of antigenic cartography maps multidimensional antigenic distances to a two-dimensional space using dimensionality reduction methods \citep{Smith:2004jc,Bedford:2014bf}.
These map-like visualizations reveal long-term patterns and trends in antigenic drift within seasonal influenza lineages like the punctuated emergence of new antigenic clusters every few years.
However, antigenic cartography is less suited to representations of short-term antigenic drift on the same time scale as annual vaccine updates in each hemisphere.

Alternate visualizations created in the nextflu \citep{NeherBedford2018} and Nextstrain \citep{Hadfield2018} frameworks address the needs of influenza researchers who make decisions about vaccine composition.
Within nextflu, all available pairwise antigenic distances between a selected reference virus's antisera and test viruses are represented by colored tips on a phylogenetic tree constructed from HA sequences (Figure~\ref{fig:1}A).
This interactive visualization allows users to select a specific reference virus by clicking a ``gear'' icon drawn on the phylogeny where the reference virus occurs.
All tips in the tree that have titer measurements to the selected reference appear as circles colored by the quantitative value of their antigenic distance (e.g., smaller distances are blue, larger distances are red).
This representation of the pairwise data allows users to identify phylogenetic clades that are antigenically distance from a given antiserum or that are missing measurements from that antiserum.
Influenza virologists use this information to select potential vaccine candidates that ``cover'' the most extant clades and prioritize which HI assays to perform next.
The benefits of visualizing the phylogenetic context of antigenic distances for a single antiserum are balanced by visual design costs of not showing antigenic distances for all antisera in a single view.

Another recent representation of pairwise antigenic distances attempts to address the limitations of the phylogenetic visualization.
This representation summarizes the mean antigenic distances between a subset of relevant reference viruses (i.e., likely vaccine candidates) and all tests viruses within each extant clade.
The resulting heatmaps use the x-axis to encode phylogenetic clades, the y-axis to encode reference viruses, and color to encode the mean $\log_{2}$ distance between a given reference virus and corresponding test viruses (Figure~\ref{fig:1}B).
These static heatmaps express more data than the phylogenetic representation, allowing decision-makers to identify qualitative trends across antisera and clades in biannual reports to the World Health Organization \citep{BedfordWHO2018,BedfordWHO2019}.
As with all heatmaps, these titer matrices suffer reduced expressiveness by encoding the most valuable quantitative data with color instead of a spatial scale.

\section{Methods}

Given the benefits and costs of existing visualizations of antigenic distances, we applied user-driven design and standard visual design principles to produce a more expressive and effective visualization for influenza virologists.
To this end, we established the goals of users who would interact with our visualizations, identified the most effective encodings for the data users needed to explore, and composed an interactive visualization from these encodings to address the desired goals.

Based on informal user interviews with collaborators at the Influenza Division of the Centers for Disease Control and Prevention, we identified a list of primary goals for the phylogenetic and heatmap visualizations.
Users wanted to know which currently circulating clades of influenza have measurements against a given serum, to prioritize which clades to select test viruses from in future HI experiments.
Additionally, users wanted to know which available antiserum has the lowest antigenic distance across all circulating clades, to identify potential vaccine candidates.
Finally, users wanted to compare the antigenic diversity of extant clades.
From these user goals, we decided that an optimal visualization would summarize the distribution of antigenic distances by antiserum and clade.

We observed that the existing titer matrix heatmaps addressed most of the user goals except for communicating which extant clades were missing measurements.
Both the phylogenetic and heatmap views use color to encode the most relevant quantitative data of antigenic distance.
Previous visualization design research has shown that quantitative data are more effectively represented by positional encodings (e.g., x- or y-axis positions) whereas nominal data (e.g., phylogenetic clades) can be effectively encoded with color \citep{Mackinlay1986}.
In the phylogenetic view, the two available positional axes are used to represent time and the unitless phylogenetic position of nodes.
Neither of these data are relevant to the user goals described above.
In the titer matrix heatmaps, the two positional axes are used to encode two nominal data types (reference virus name and clade name).

We reasoned that we could make a more effective visualization that addressed most user goals by changing the encoding of data in the titer matrix heatmaps.
Specifically, we chose to temporally omit clade names from the visualization and encode the antigenic distances on the positional x-axis.
By encoding antigenic distance on a positional axis, we could annotate relevant thresholds for antigenic distances, show all available measurements for each reference virus at once, and display a summary statistic (mean antigenic distance) for each reference virus.
We chose to maintain the encoding of nominal reference virus names on the y-axis, since most user goals require interrogation of specific antisera.
With color available as an additional channel, we decided to encode the antigenic class of each antigenic distance with color.
We implemented this design as a new interactive panel in Nextstrain's visualization tool, Auspice (Figure?).

\section{Results}

\subsection{Interactive visualization of titer measurements}

\subsection{Case study 1: Retrospective vaccine strain selection in Fall 2009}

Before and after vaccine strain selection (Figure~\ref{fig:2}).

\subsection{Case study 2: Identification of genotype-specific patterns through visualization of raw data}

Visualization of raw data instead of summary statistics reveals different distributions that correspond to genotype-specific titer measurements (Figure~\ref{fig:3}).

\section{Discussion}

\section*{Conflict of Interest Statement}
%All financial, commercial or other relationships that might be perceived by the academic community as representing a potential conflict of interest must be disclosed. If no such relationship exists, authors will be asked to confirm the following statement:

The authors declare that the research was conducted in the absence of any commercial or financial relationships that could be construed as a potential conflict of interest.

\section*{Author Contributions}

JL, TRS, TB, and JHa designed and implemented measurements panel in Auspice.
JHu and AB designed and implemented initial prototypes of interactive measurements visualizations.
JHu, AB, and RAN designed and implemented static visualizations.
JL, JHu, and JHa wrote manuscript.

\section*{Funding}
Details of all funding sources should be provided, including grant numbers if applicable. Please ensure to add all necessary funding information, as after publication this is no longer possible.

\section*{Acknowledgments}
This is a short text to acknowledge the contributions of specific colleagues, institutions, or agencies that aided the efforts of the authors.

\section*{Supplemental Data}
 \href{http://home.frontiersin.org/about/author-guidelines#SupplementaryMaterial}{Supplementary Material} should be uploaded separately on submission, if there are Supplementary Figures, please include the caption in the same file as the figure. LaTeX Supplementary Material templates can be found in the Frontiers LaTeX folder.

\section*{Data Availability Statement}
The datasets [GENERATED/ANALYZED] for this study can be found in the [NAME OF REPOSITORY] [LINK].
% Please see the availability of data guidelines for more information, at https://www.frontiersin.org/about/author-guidelines#AvailabilityofData

\bibliographystyle{Frontiers-Harvard}
\bibliography{measurements-panel}

%%% Make sure to upload the bib file along with the tex file and PDF
%%% Please see the test.bib file for some examples of references

\section*{Figure captions}

%%% Please be aware that for original research articles we only permit a combined number of 15 figures and tables, one figure with multiple subfigures will count as only one figure.
%%% Use this if adding the figures directly in the mansucript, if so, please remember to also upload the files when submitting your article
%%% There is no need for adding the file termination, as long as you indicate where the file is saved. In the examples below the files (logo1.eps and logos.eps) are in the Frontiers LaTeX folder
%%% If using *.tif files convert them to .jpg or .png
%%%  NB logo1.eps is required in the path in order to correctly compile front page header %%%

\begin{figure}[h!]
  \begin{center}
    \includegraphics[width=\textwidth]{figures/figure-1-mockup-static-visualizations-of-serological-data-for-decision-makers}
  \end{center}
  \caption{
    Previous approaches to static visualization of serological data for seasonal influenza vaccine composition reports.
A) Phylogenetic visualization \citep{NeherBedford2018} allows the user to select a single vaccine candidate (e.g., A/Texas/50/2012) and see how well that strain might protect against other circulating strains in their genetic context based on the serological distance encoded by color (orange and red color indicate greater distance and less protection by the selected strain).
To compare multiple vaccine candidates, users have to select different strains manually and toggle between them.
B) Heatmap visualization of mean serological distances between multiple vaccine candidates (reference strains on the y-axis) and strains in currently circulating phylogenetic clades.
Heatmaps encode distance by color and display the distance as text, allowing the user to compare how well multiple vaccine candidates might protect against circulating strains.
C) Interval plot of mean +/- 89\% confidence interval values of serological distances between vaccine candidates (y-axis) and strains in currently circulating clades.
Unlike the heatmap visualization, the interval plot encodes serological distance with a spatial scale (the x-axis) instead of color and encodes clade membership with color instead of the spatial scale.
The vertical gray lines represent the threshold above which strains are considered antigenically distinct (x=2, solid line) and where strains are antigenically identical (x=0, dashed line).
This view allows users to compare multiple vaccine candidates, identify the candidate that protects specific clades based on a mean value to the left of the threshold at x=2, and view the variance in the underlying serological measurements.
D) Combined swarm and interval plot showing the raw pairwise measurements between each vaccine candidate and the test strains in each clade.
This view allows users to perform the same tasks as the interval plot, but it also allows users to identify how many measurements support the summary statistics for a given vaccine candidate and identify multiple modes in the raw data distribution that could indicate within-clade antigenic variation.
}\label{fig:1}
\end{figure}

\setcounter{figure}{2}
\setcounter{subfigure}{0}
\begin{subfigure}
\setcounter{figure}{2}
\setcounter{subfigure}{0}
    \centering
    \begin{minipage}[b]{0.75\textwidth}
        \includegraphics[width=\linewidth]{figures/figure-2a-mockup}
        \caption{Before vaccine selection.}
        \label{fig:2A}
    \end{minipage}

\setcounter{figure}{2}
\setcounter{subfigure}{1}
    \begin{minipage}[b]{0.75\textwidth}
        \includegraphics[width=\linewidth]{figures/figure-2b-mockup}
        \caption{After vaccine selection.}
        \label{fig:2B}
    \end{minipage}

\setcounter{figure}{2}
\setcounter{subfigure}{-1}
    \caption{Enter the caption for your subfigure here. \textbf{(A)} This is the caption for Subfigure 1. \textbf{(B)} This is the caption for Subfigure 2.}
    \label{fig:2}
\end{subfigure}

\begin{figure}[h!]
  \begin{center}
    \includegraphics[width=\textwidth]{figures/figure-3-mockup-for-case-study-2}
  \end{center}
  \caption{
    Visualization of raw data and summary statistics.
A) Phylogenetic tree.
B) Titer measurements summarized by reference strain and test strain clade with mean and standard deviation.
C) Individual (raw) titer measurements showing pairwise values between test strains (colored by clade) and the reference strains showing a previously hidden bimodal distribution in the measurements for the A/Shanghai/11/1987 reference strain compared to a unimodal distribution for the A/Beijing/353/1989 reference strain.
D) Individual titer measurements colored by genotype at site HA1:135, revealing a potential genotype-specific explanation for the two clusters seen in the A/Shanghai/11/1987 measurements and a similar bimodal distribution in the measurements for A/Beijing/353/1989 that was not as clear when coloring raw measurements by clade alone.}\label{fig:3}
\end{figure}

%%% If you don't add the figures in the LaTeX files, please upload them when submitting the article.
%%% Frontiers will add the figures at the end of the provisional pdf automatically
%%% The use of LaTeX coding to draw Diagrams/Figures/Structures should be avoided. They should be external callouts including graphics.

\end{document}
